### PR TITLE
adopt GH dependabot

### PR DESCRIPTION
**Pull request by bjlittle**
### Description
This PR enables GH `dependabot`, which will generate automated pull-requests with GHA version updates.

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

**Comment by bjlittle** [comment]
@trexfeathers Benchmarking seems broken... don't quite understand why.

Any ideas? :thinking:

File: docs/src/conf.py [hunk 1]
```diff
@@ -228,7 +228,7 @@ def _dotv(version):
     "matplotlib": ("https://matplotlib.org/stable/", None),
     "numpy": ("https://numpy.org/doc/stable/", None),
     "python": ("https://docs.python.org/3/", None),
-    "scipy": ("https://docs.scipy.org/doc/scipy/", None),
+    "scipy": ("https://docs.scipy.org/doc/", None),
```
Comment (bjlittle) [inline docs/src/conf.py:231]: `scipy` 1.8.0 was released yesterday (sunday 06 feb 2022) so i guess some stuff moved under the hood for their docs

Comment (bjlittle) [inline docs/src/conf.py:231]: `scipy` 1.8.0 was released yesterday (sunday 06/02/2022) and there seems to be some issues with their documentation, see https://github.com/scipy/scipy/issues/15540

**Comment by trexfeathers** [comment]
> @trexfeathers Benchmarking seems broken... don't quite understand why.
> 
> Any ideas? 🤔

ASV released a new version recently, and something we're subclassing now needs an extra argument. That's an easy fix, but I'm just looking at whether I can make things a bit more robust at the same time...

**Comment by bjlittle** [comment]
@trexfeathers Awesome, thanks!

Let's see how `dependabot` behaves with regards to third-party GHA and explicit commit hashes :eyes:

**Pull request merged**